File: CC_C#/BVCS Projects/Blackboard/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace Blackboard
{
    partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        bool mousePress;
        int xLast;
        int yLast;
        Graphics myGraphics;
        Pen myPen;

        private void Form1_Load(object sender, EventArgs e)
        {
            // Create graphics and pen objects
            myGraphics = pnlBlackboard.CreateGraphics();
            myPen = new Pen(Color.Gray, 1);
            // Initialize the eight radio button colors 
            rdoGray.BackColor = Color.Gray;
            rdoBlue.BackColor = Color.Blue;
            rdoGreen.BackColor = Color.LightGreen;
            rdoCyan.BackColor = Color.Cyan;
            rdoRed.BackColor = Color.Red;
            rdoMagenta.BackColor = Color.Magenta;
            rdoYellow.BackColor = Color.Yellow;
            rdoWhite.BackColor = Color.White;
            // Set initial color
            rdoGray.Checked = true;
            mousePress = false;
        }

        private void rdoGray_CheckedChanged(object sender, EventArgs e)
        {
            // Gray 
            myPen.Color = rdoGray.BackColor;
        }

        private void rdoBlue_CheckedChanged(object sender, EventArgs e)
        {
            // Blue 
            myPen.Color = rdoBlue.BackColor;
        }

        private void rdoGreen_CheckedChanged(object sender, EventArgs e)
        {
            // Green 
            myPen.Color = rdoGreen.BackColor;
        }

        private void rdoCyan_CheckedChanged(object sender, EventArgs e)
        {
            // Cyan
            myPen.Color = rdoCyan.BackColor;
        }

        private void rdoRed_CheckedChanged(object sender, EventArgs e)
        {
            // Red 
            myPen.Color = rdoRed.BackColor;
        }

        private void rdoMagenta_CheckedChanged(object sender, EventArgs e)
        {
            // Magenta
            myPen.Color = rdoMagenta.BackColor;
        }

        private void rdoYellow_CheckedChanged(object sender, EventArgs e)
        {
            // Yellow
            myPen.Color = rdoYellow.BackColor;
        }

        private void rdoWhite_CheckedChanged(object sender, EventArgs e)
        {
            // White
            myPen.Color = rdoWhite.BackColor;
        }

        private void btnErase_Click(object sender, EventArgs e)
        {
            // Clear the blackboard
            myGraphics.Clear(pnlBlackboard.BackColor);
        }

        private void btnExit_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void pnlBlackboard_MouseDown(object sender, MouseEventArgs e)
        {
            // Start drawing if left click
            if (e.Button == MouseButtons.Left)
            {
                mousePress = true;
                xLast = e.X;
                yLast = e.Y;
            }
        }

        private void pnlBlackboard_MouseMove(object sender, MouseEventArgs e)
        {
            // Draw a line if drawing
            if (mousePress)
            {
                myGraphics.DrawLine(myPen, xLast, yLast, e.X, e.Y);
                xLast = e.X;
                yLast = e.Y;
            }
        }

        private void pnlBlackboard_MouseUp(object sender, MouseEventArgs e)
        {
            // Finish line
            if (e.Button == MouseButtons.Left)
            {
                myGraphics.DrawLine(myPen, xLast, yLast, e.X, e.Y);
                mousePress = false;
            }
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e)
        {
            myGraphics.Dispose();
            myPen.Dispose();
        }
    }
}

File: CC_Visual Basic/VisualBasic/BeginVBE/BeginVBE/BVBE Projects/Blackboard/Form1.vb
Public Class Form1
  Dim MousePress As Boolean
  Dim XLast As Integer, YLast As Integer
  Dim MyGraphics As Graphics
  Dim MyPen As Pen

  Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
    MyGraphics.Dispose()
    MyPen.Dispose()
  End Sub

  Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
    'Create graphics and pen objects
    MyGraphics = pnlBlackboard.CreateGraphics
    MyPen = New Pen(Color.Gray, 1)
    'Initialize the eight radio button colors 
    rdoGray.BackColor = Color.Gray
    rdoBlue.BackColor = Color.Blue
    rdoGreen.BackColor = Color.LightGreen
    rdoCyan.BackColor = Color.Cyan
    rdoRed.BackColor = Color.Red
    rdoMagenta.BackColor = Color.Magenta
    rdoYellow.BackColor = Color.Yellow
    rdoWhite.BackColor = Color.White
    'Set initial color
    rdoGray.Checked = True
    MousePress = False
  End Sub

  Private Sub rdoGray_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoGray.CheckedChanged
    'Gray 
    MyPen.Color = rdoGray.BackColor
  End Sub

  Private Sub rdoBlue_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoBlue.CheckedChanged
    'Blue
    MyPen.Color = rdoBlue.BackColor
  End Sub

  Private Sub rdoGreen_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoGreen.CheckedChanged
    'Green
    MyPen.Color = rdoGreen.BackColor
  End Sub

  Private Sub rdoCyan_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoCyan.CheckedChanged
    'Cyan
    MyPen.Color = rdoCyan.BackColor
  End Sub

  Private Sub rdoRed_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoRed.CheckedChanged
    'Red 
    MyPen.Color = rdoRed.BackColor
  End Sub

  Private Sub rdoMagenta_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoMagenta.CheckedChanged
    'Magenta 
    MyPen.Color = rdoMagenta.BackColor
  End Sub

  Private Sub rdoYellow_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoYellow.CheckedChanged
    'Yellow
    MyPen.Color = rdoYellow.BackColor
  End Sub

  Private Sub rdoWhite_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoWhite.CheckedChanged
    'White
    MyPen.Color = rdoWhite.BackColor
  End Sub

  Private Sub btnErase_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnErase.Click
    'Clear the blackboard
    MyGraphics.Clear(pnlBlackboard.BackColor)
  End Sub

  Private Sub btnExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnExit.Click
    Me.Close()
  End Sub

  Private Sub pnlBlackboard_MouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles pnlBlackboard.MouseDown
    'Start drawing if left click
    If e.Button = MouseButtons.Left Then
      MousePress = True
      XLast = e.X
      YLast = e.Y
    End If
  End Sub

  Private Sub pnlBlackboard_MouseMove(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles pnlBlackboard.MouseMove
    'Draw a line if drawing
    If MousePress Then
      MyGraphics.DrawLine(MyPen, XLast, YLast, e.X, e.Y)
      XLast = e.X
      YLast = e.Y
    End If
  End Sub

    Private Sub pnlBlackboard_MouseUp(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles pnlBlackboard.MouseUp
        'Finish line
        If e.Button = MouseButtons.Left Then
            MyGraphics.DrawLine(MyPen, XLast, YLast, e.X, e.Y)
            MousePress = False
        End If
    End Sub
End Class
